src/economy.cpp

```diff
@@ -248,7 +248,7 @@ int UpdateCompanyRatingAndValue(Company *c, bool update)
 	{
 		int numec = std::min<uint>(c->num_valid_stat_ent, 12u);
 		if (numec != 0) {
-			const CompanyEconomyEntry *cee = c->old_economy;
+			const CompanyEconomyEntry *cee = &c->old_economy[0];
```

Comment:
i feel like it would be better to fix usages of these sort of "iterators" at the same time as converting to array. Unless that's coming next anyway of course :)

Reply:
Good point, I should've converted it to iterators. It's a bit weird as it doesn't check for `.end()`, but that's not needed because we'll stop before that. Added a few `static_assert`s to make that a bit clearer though.

Reply:
i kind meant rework the loop to not need to use iterators, but this is still better :)

Reply:
What about the new incarnation? C++ enough? :D

Reply:
Wild